homeassistant/components/niko_home_control/light.py

```diff
@@ -16,29 +16,87 @@
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN, HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
 from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import issue_registry as ir
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 from homeassistant.util import Throttle
 
+from .const import DOMAIN
+
 _LOGGER = logging.getLogger(__name__)
 MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=1)
 SCAN_INTERVAL = timedelta(seconds=30)
 
+# delete after 2025.5.0
 PLATFORM_SCHEMA = LIGHT_PLATFORM_SCHEMA.extend({vol.Required(CONF_HOST): cv.string})
 
 
 async def async_setup_platform(
     hass: HomeAssistant,
-    config: ConfigType,
+    hass_config: ConfigType,
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the light platform."""
+
+    if hass_config.get("platform") == DOMAIN:
+        hass.async_create_task(_async_import(hass, hass_config))
+
+
+# delete after 2025.5.0
+async def _async_import(hass: HomeAssistant, config: ConfigType) -> None:
+    """Set up the nhc environment."""
+    if not hass.config_entries.async_entries(DOMAIN):
+        # Start import flow
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+        )
+        if result.get("type") == FlowResultType.ABORT:
+            ir.async_create_issue(
+                hass,
+                DOMAIN,
+                f"deprecated_yaml_import_issue_{result.get('reason', 'unknown')}",
+                breaks_in_ha_version="2025.5.0",
+                is_fixable=False,
+                issue_domain=DOMAIN,
+                severity=ir.IssueSeverity.WARNING,
+                translation_key=f"deprecated_yaml_import_issue_{result.get('reason', 'unknown')}",
+                translation_placeholders={
+                    "domain": DOMAIN,
+                    "integration_title": "niko_home_control",
+                },
+            )
+            return
+
+    ir.async_create_issue(
+        hass,
+        HOMEASSISTANT_DOMAIN,
+        f"deprecated_yaml_{DOMAIN}",
+        breaks_in_ha_version="2025.5.0",
+        is_fixable=False,
+        issue_domain=DOMAIN,
+        severity=ir.IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+        translation_placeholders={
+            "domain": DOMAIN,
+            "integration_title": "niko_home_control",
+        },
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the Niko Home Control light platform."""
-    host = config[CONF_HOST]
+    host = hass.data[DOMAIN][entry.entry_id]["config"]["host"]
 
     try:
         nhc = nikohomecontrol.NikoHomeControl(
```

Comment:
Why don't we create this object in `__init__` and store that one in `entry.runtime_data`? I assume this object is also relevant for other platforms

Reply:
I think because the docs said to do it that way 🤣

Reply:
What docs

Reply:
Exactly!